distributed/shuffle/_core.py

```diff
@@ -0,0 +1,278 @@
+from __future__ import annotations
+
+import abc
+import asyncio
+import contextlib
+import itertools
+import time
+from collections import defaultdict
+from collections.abc import Callable, Iterator
+from concurrent.futures import ThreadPoolExecutor
+from dataclasses import dataclass, field
+from enum import Enum
+from typing import TYPE_CHECKING, Any, ClassVar, Generic, NewType, TypeVar
+
+import pandas as pd
```

Comment:
Previously this was guarded by `if TYPE_CHECKING` which looks to be the cause of most of the test failures.

I note `get_output_partition` advertises that it accepts `meta : pd.DataFrame | None`. Probably that should be `meta: _T_partition_type | None` and then the pandas import can go from this generic core.

Reply:
Thanks, I just looked at the test report and missed the complete failure of `mindeps`.